src/tests/JIT/SIMD/ShiftOperations.cs

```diff
@@ -0,0 +1,487 @@
+using System;
+using System.Runtime.CompilerServices;
+using System.Numerics;
+
+public class Test
+{	
+	 
+	[MethodImpl(MethodImplOptions.NoInlining)]
+    private static uint Shlx32bit(uint x, int y) => x<< y;
+	
+	[MethodImpl(MethodImplOptions.NoInlining)]
+    private static int Sarx32bit(int x, int y) => x >> y;
+ 
+    [MethodImpl(MethodImplOptions.NoInlining)]
+    private static uint Shrx32bit(uint x, int y) => x >> y;
+	
+	[MethodImpl(MethodImplOptions.NoInlining)]
+    private static uint Ror(uint x) => BitOperations.RotateRight(x, 2);
+
+  	[MethodImpl(MethodImplOptions.NoInlining)]
+    private static ulong Shlx64bit(ulong x, int y) => x << y;
+
+	[MethodImpl(MethodImplOptions.NoInlining)]
+    private static long Sarx64bit(long x, int y) => x >> y;
+
+    [MethodImpl(MethodImplOptions.NoInlining)]
+    private static ulong Shrx64bit(ulong x, int y) => x >> y;
+
+	[MethodImpl(MethodImplOptions.NoInlining)]
+    private static unsafe ulong ShrxRef64bit(ulong *x, int y) => *x >> y;
+		
+ 	[MethodImpl(MethodImplOptions.NoInlining)]
+    private static ulong Rorx(ulong x) => BitOperations.RotateRight(x, 2);
+	
+public static unsafe int Main()
+{
+	try
+	{
+		uint  valUInt = 0;
+		int   valInt = 0;		
+		ulong valULong = 0;
+		long  valLong = 0;
+		int   shiftBy = 0;		
+		uint  resUInt = 0;
+		int   resInt = 0;
+		ulong resULong = 0;
+		long  resLong = 0;
+		uint  expectedUInt = 0;
+		int   expectedInt = 0;
+		ulong expectedULong = 0;
+		long  expectedLong = 0;
+		int   MOD32 = 32;
+		int   MOD64 = 64;
+		
+		//
+		// Shlx32bit tests
+		//
+		valUInt = 0;
+		shiftBy = 1;
+		resUInt = Shlx32bit(valUInt, shiftBy);
+		expectedUInt = (uint) (valUInt * Math.Pow(2, (shiftBy % MOD32)));
+		Console.Write("UnitTest Shlx32bit({0},{1}): {2}", valUInt, shiftBy, resUInt);
+		if (resUInt != expectedUInt)
+		{
+			Console.Write(" != {0} Failed.\n", expectedUInt);
+			return 101;
```

Comment:
You should consider letting all the tests run, and not exiting on the first failure. Just make sure to return 101 if *any* test fails.

Reply:
Done.